packages/contracts-bedrock/src/L1/ResourceMetering.sol

```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.15;
+pragma solidity ^0.8.15;
```

Comment:
Are these pragma changes intentional/necessary?

Reply:
I don't believe we want to change the pragma changes here altho resource metering prob could be `^0.8.0` but we shouldn't change that as part of this

Reply:
unfortunately these version changes are intentional, since they are dependencies of contract `EIP1967Helper`, which we are newly importing in `Deployer.sol` to use the library in `loadInitializedSlot`: https://github.com/ethereum-optimism/optimism/blob/85b5949325acc4dd7c7e85531de8336d70ae0c81/packages/contracts-bedrock/scripts/Deployer.sol#L10

If we leave these dependencies, such as `ResourceMetering.sol`, unchanged, we run into the following error at compilation:

```
[⠃] Compiling...2024-01-08T22:10:45.401194Z ERROR foundry_compilers::resolver: failed to resolve versions
[⠊] Compiling...
Error: 
Found incompatible Solidity versions:
scripts/DeployL2.s.sol (^0.8.0) imports:
    scripts/Deployer.sol (^0.8.0)
    scripts/DeployConfig.s.sol (^0.8.0)
    lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    src/EAS/EAS.sol (0.8.19)
    src/EAS/SchemaRegistry.sol (0.8.19)
    src/EAS/ISchemaRegistry.sol (^0.8.0)
    src/libraries/Predeploys.sol (^0.8.0)
    lib/forge-std/src/Script.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdJson.sol (>=0.6.0 <0.9.0)
    lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    scripts/Executables.sol (^0.8.0)
    scripts/Chains.sol (^0.8.0)
    src/libraries/Predeploys.sol (^0.8.0)
    test/mocks/EIP1967Helper.sol (^0.8.0)
    src/legacy/AddressManager.sol (^0.8.15)
    lib/solady/src/utils/LibString.sol (^0.8.4)
    lib/forge-std/src/Script.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    lib/forge-std/src/StdJson.sol (>=0.6.0 <0.9.0)
    scripts/Executables.sol (^0.8.0)
    scripts/Chains.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Address.sol (^0.8.1)
    src/universal/ISemver.sol (^0.8.0)
    src/libraries/Predeploys.sol (^0.8.0)
    src/EAS/eip1271/EIP1271Verifier.sol (0.8.19)
    src/EAS/resolver/ISchemaResolver.sol (^0.8.0)
    src/EAS/Common.sol (^0.8.0)
    src/EAS/IEAS.sol (^0.8.0)
    src/EAS/ISchemaRegistry.sol (^0.8.0)
    src/universal/ISemver.sol (^0.8.0)
    src/EAS/resolver/ISchemaResolver.sol (^0.8.0)
    src/EAS/Common.sol (^0.8.0)
    src/EAS/ISchemaRegistry.sol (^0.8.0)
    src/EAS/resolver/ISchemaResolver.sol (^0.8.0)
    lib/forge-std/src/console.sol (>=0.4.22 <0.9.0)
    lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    lib/forge-std/src/safeconsole.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdChains.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdCheats.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdJson.sol (>=0.6.0 <0.9.0)
    lib/forge-std/src/StdMath.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdStorage.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdStyle.sol (>=0.4.22 <0.9.0)
    lib/forge-std/src/StdUtils.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Base.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    src/libraries/Constants.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/access/Ownable.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/cryptography/draft-EIP712.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/cryptography/SignatureChecker.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Address.sol (^0.8.1)
    src/EAS/IEAS.sol (^0.8.0)
    src/EAS/Common.sol (^0.8.0)
    src/EAS/Common.sol (^0.8.0)
    src/EAS/ISchemaRegistry.sol (^0.8.0)
    src/EAS/Common.sol (^0.8.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdStorage.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/interfaces/IMulticall3.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/mocks/MockERC20.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/mocks/MockERC721.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/StdStorage.sol (>=0.6.2 <0.9.0)
    lib/forge-std/src/Vm.sol (>=0.6.2 <0.9.0)
    src/L1/ResourceMetering.sol (0.8.15)
    lib/openzeppelin-contracts/contracts/utils/Context.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Address.sol (^0.8.1)
    lib/openzeppelin-contracts/contracts/interfaces/IERC1271.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/proxy/utils/Initializable.sol (^0.8.2)
    lib/openzeppelin-contracts/contracts/utils/math/Math.sol (^0.8.0)
    src/libraries/Burn.sol (^0.8.15)
    src/libraries/Arithmetic.sol (^0.8.15)
    lib/openzeppelin-contracts/contracts/utils/Strings.sol (^0.8.0)
    lib/openzeppelin-contracts/contracts/utils/Address.sol (^0.8.1)
    lib/openzeppelin-contracts/contracts/utils/math/SignedMath.sol (^0.8.0)
    lib/solmate/src/utils/FixedPointMathLib.sol (>=0.8.0)
 ELIFECYCLE  Test failed. See above for more details.
```

have any thoughts?

Reply:
fwiw, im hoping for an alternative, as this would have a lot of side effects on the codebase

Reply:
This is a fine solution for now